scipy/stats/_variation.py

```diff
@@ -0,0 +1,211 @@
+    # to propagate to the user.  Unfortunately, there does not
+    # appear to be a thread-safe way to filter out the warnings,
+    # so we have to do the calculation in a way that doesn't
+    # generate numpy warnings.
+    #
+    # Let N be the number of non-nan inputs in a slice.
+    # Conditions that generate nan:
+    #   * empty input (i.e. N = 0)
+    #   * All non-nan values 0
+    #   * N < ddof
+    #   * N == ddof and the input is constant
+    # Conditions that generate inf:
+    #   * non-constant input and either
+    #       * the mean is 0, or
+    #       * N == ddof
+    #
+    a_isnan = np.isnan(a)
+    all_nan = a_isnan.all(axis=axis, keepdims=True)
+    all_nan_full = np.broadcast_to(all_nan, a.shape)
+    all_zero = (a_isnan | (a == 0)).all(axis=axis, keepdims=True) & ~all_nan
+
+    # ngood is the number of non-nan values in each slice.
+    ngood = (a.shape[axis] -
+             np.expand_dims(np.count_nonzero(a_isnan, axis=axis), axis))
+    # The return value is nan where ddof > ngood.
+    ddof_too_big = ddof > ngood
+    # If ddof == ngood, the return value is nan where the input is constant and
+    # inf otherwise.
+    ddof_equal_n = ddof == ngood
+
+    is_const = _nan_allsame(a, axis=axis, keepdims=True)
+
+    a2 = a.copy()
+    # If an entire slice is nan, `np.nanmean` will generate a warning,
+    # so we replace those nan's with 1.0 before computing the mean.
+    # We'll fix the corresponding output later.
+    a2[all_nan_full] = 1.0
+    mean_a = np.nanmean(a2, axis=axis, keepdims=True)
+
+    # If ddof >= ngood (the number of non-nan values in the slice), `np.nanstd`
+    # will generate a warning, so set all the values in such a slice to 1.0.
+    # We'll fix the corresponding output later.
+    a2[np.broadcast_to(ddof_too_big, a2.shape) | ddof_equal_n] = 1.0
+    std_a = np.nanstd(a2, axis=axis, ddof=ddof, keepdims=True)
+    del a2
+
+    sum_zero = np.nansum(a, axis=axis, keepdims=True) == 0
+
+    # Where the sum along the axis is 0, replace mean_a with 1.  This avoids
+    # division by zero.  We'll fix the the corresponding output later.
+    mean_a[sum_zero] = 1.0
+
+    # Here--finally!--is the calculation of the variation.
+    result = std_a / mean_a
+
+    # Now fix the values that were given fake data to avoid warnings.
+    result[~is_const & sum_zero] = np.inf
+    signed_inf_mask = ~is_const & ddof_equal_n
+    result[signed_inf_mask] = np.sign(mean_a[signed_inf_mask]) * np.inf
+    nan_mask = all_zero | all_nan | ddof_too_big | (ddof_equal_n & is_const)
+    result[nan_mask] = np.nan
+
+    if not keepdims:
+        result = np.squeeze(result, axis=axis)
+        if result.shape == ():
+            result = result[()]
+
+    return result
+
+
+def variation(a, axis=0, nan_policy='propagate', *, ddof=0, keepdims=False):
+    """
+    Compute the coefficient of variation.
+
+    The coefficient of variation is the ratio of the standard deviation
+    to the mean.
```

Comment:
is it worthwhile to write the eqn here to resolve any ambiguity on the reader/user?

Reply:
Sure, I'll add that in an update.

Reply:
Since this comment was made, the docstring had been updated with the formula (expressed with numpy function calls). I've updated the docstring in the new file with the same changes.